From: Alexander Mikhaylenko <alexander.mikhaylenko@puri.sm>
Date: Thu, 24 Sep 2020 12:06:38 +0500
Subject: librem5: webapp-additional-urls-dialog: Slightly modernize the
 layout

Drop the icon and move Clear button to the bottom.

Upstreamed in https://gitlab.gnome.org/GNOME/epiphany/-/merge_requests/834
---
 src/resources/gtk/webapp-additional-urls-dialog.ui | 79 +++++++---------------
 1 file changed, 23 insertions(+), 56 deletions(-)

diff --git a/src/resources/gtk/webapp-additional-urls-dialog.ui b/src/resources/gtk/webapp-additional-urls-dialog.ui
index 0ee6475..8564cbc 100644
--- a/src/resources/gtk/webapp-additional-urls-dialog.ui
+++ b/src/resources/gtk/webapp-additional-urls-dialog.ui
@@ -8,79 +8,29 @@
     </columns>
   </object>
   <template class="EphyWebappAdditionalURLsDialog" parent="GtkDialog">
-    <property name="height_request">500</property>
     <property name="modal">True</property>
-    <property name="window_position">center</property>
+    <property name="window_position">center-on-parent</property>
     <property name="default_width">300</property>
     <property name="default_height">400</property>
     <property name="destroy_with_parent">True</property>
     <property name="type_hint">dialog</property>
-    <child internal-child="headerbar">
-      <object class="GtkHeaderBar">
-        <property name="title" translatable="yes">Additional URLs</property>
-        <property name="show-close-button">True</property>
-        <child>
-          <object class="GtkButton">
-            <property name="label" translatable="yes">C_lear All</property>
-            <property name="visible">True</property>
-            <property name="use_underline">True</property>
-            <property name="valign">center</property>
-            <property name="action-name">webapp-additional-urls.forget-all</property>
-            <accelerator key="Delete" modifiers="GDK_SHIFT_MASK" signal="clicked"/>
-            <style>
-              <class name="destructive-action"/>
-              <class name="text-button"/>
-            </style>
-          </object>
-        </child>
-      </object>
-    </child>
+    <property name="title" translatable="yes">Additional URLs</property>
     <child internal-child="vbox">
       <object class="GtkBox">
         <child>
-          <object class="GtkBox">
+          <object class="GtkLabel">
             <property name="visible">True</property>
-            <property name="can_focus">False</property>
+            <property name="halign">start</property>
             <property name="margin">12</property>
-            <child>
-              <object class="GtkImage">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="icon_name">dialog-information-symbolic</property>
-                <property name="icon_size">6</property>
-                <property name="margin_left">6</property>
-                <property name="margin_right">6</property>
-              </object>
-              <packing>
-                <property name="expand">False</property>
-                <property name="fill">True</property>
-                <property name="position">0</property>
-              </packing>
-            </child>
-            <child>
-              <object class="GtkLabel">
-                <property name="visible">True</property>
-                <property name="halign">start</property>
-                <property name="label" translatable="yes">A URL that starts with any of the additional URLs will be opened by the web application. If you omit the URL scheme, the one from the currently loaded URL will be used.</property>
-                <property name="wrap">True</property>
-              </object>
-              <packing>
-                <property name="expand">True</property>
-                <property name="fill">True</property>
-                <property name="position">1</property>
-              </packing>
-            </child>
+            <property name="label" translatable="yes">A URL that starts with any of the additional URLs will be opened by the web application. If you omit the URL scheme, the one from the currently loaded URL will be used.</property>
+            <property name="wrap">True</property>
           </object>
         </child>
         <child>
           <object class="GtkScrolledWindow">
-            <property name="width_request">400</property>
-            <property name="height_request">300</property>
             <property name="visible">True</property>
             <property name="expand">True</property>
             <property name="hscrollbar_policy">never</property>
-            <property name="min_content_width">300</property>
-            <property name="min_content_height">300</property>
             <child>
               <object class="GtkTreeView" id="treeview">
                 <property name="visible">True</property>
@@ -146,6 +96,23 @@
                 </child>
               </object>
             </child>
+            <child>
+              <object class="GtkButton">
+                <property name="label" translatable="yes">C_lear All</property>
+                <property name="visible">True</property>
+                <property name="use_underline">True</property>
+                <property name="valign">center</property>
+                <property name="action-name">webapp-additional-urls.forget-all</property>
+                <accelerator key="Delete" modifiers="GDK_SHIFT_MASK" signal="clicked"/>
+                <style>
+                  <class name="destructive-action"/>
+                  <class name="text-button"/>
+                </style>
+              </object>
+              <packing>
+                <property name="pack-type">end</property>
+              </packing>
+            </child>
           </object>
         </child>
       </object>
